### PR TITLE
Update detekt Config, README, gradle version etc...

### PR DESCRIPTION
**Pull request by sleepylee**
## What happened 🤔
Some minor update on the static code analysis we are using: `detekt`

## Insight 👀
- Update detekt config, use newer version
- Update README

## POW
image

### CODE REVIEW

File: detekt-config.yml [hunk 1]
```diff
@@ -1,24 +1,3 @@
...
-    - 'comments'
-  exclude-rules:
-    - 'NamingRules'
-    - 'WildcardImport'
-    - 'MagicNumber'
-    - 'MaxLineLength'
-    - 'LateinitUsage'
-    - 'StringLiteralDuplication'
-    - 'SpreadOperator'
-    - 'TooManyFunctions'
```
Comment (luongvo) [inline detekt-config.yml:20]: 👍

File: build.gradle [hunk 2]
```diff
@@ -78,13 +74,21 @@ task clean(type: Delete) {
...
-        output = "${project.buildDir}/reports/detekt/"
-        parallel = true
-        disableDefaultRuleSets = false
-    }
+configurations {
+    detekt
+ }
+
+dependencies {
+    detekt "io.gitlab.arturbosch.detekt:detekt-cli:1.12.0"
```
Comment (luongvo) [inline build.gradle:82]: Why do we switch to use detekt-cli? 🤔 
Also switch from using `detekt` block to a custom task?

Comment (sleepylee) [reply build.gradle:82]: creating a task is easier so you can still run `./gradlew detekt` but also you can run it directly from the gradle file, it looks like this (just like task :clean we had before):
image

Comment (sleepylee) [reply build.gradle:82]: I'll recheck whether `detekt-cli` is necessary 🤔  from a setup tutorial I think they used it and I didn't realize that 🤔

Comment (sleepylee) [reply build.gradle]: updated 🙏

**Review by ViktorArtemiev — DISMISSED** [review]

File: build.gradle [hunk 3]
```diff
@@ -62,7 +62,11 @@ buildscript {
 }
 
 plugins {
-    id "io.gitlab.arturbosch.detekt" version "1.0.0.RC6-3"
+    id "io.gitlab.arturbosch.detekt" version "1.14.0"
+}
+
+repositories {
+    jcenter()
```
Comment (luongvo) [inline build.gradle:69]: @sleepylee do we still need this config as we had it in project gradle already?

Comment (sleepylee) [reply build.gradle:69]: I followed their instruction to have a top-level `repositories`🤔  maybe we can try to remove it 
image

**Review by manh-t — APPROVED** [review]

File: build.gradle (hunk 3 above)
Comment (luongvo) [reply build.gradle:69]: @sleepylee so, will you remote it? 🤔

Comment (sleepylee) [reply build.gradle:69]: yes we can remove it 💯 confirmed and will commit soon

**Comment by sleepylee** [comment]
updated, please review again @nimblehq/android-chapter

**Review by manh-t — APPROVED** [review]

**Review by luongvo — APPROVED** [review]

**Pull request merged**